### PR TITLE
Twitter demo in Java.

### PR DESCRIPTION
**Pull request by artursouza**
Twitter demo in Java.

image

### CODE REVIEW

File: twitter-sentiment-processor/demos/javademo/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+```
+
+## Undersanding the Pipeline
+
+All the applications described here uses Spring Boot.
+
+### Provider
+
+It will receive Twitter feeds, invoke the processor application above, then save the results to state store and finally publishes to a topic. This application uses multiple Dapr building blocks:
```
Comment (mchmarny) [inline twitter-sentiment-processor/demos/javademo/README.md:58]: What is "it" in this sentence?

File: twitter-sentiment-processor/demos/javademo/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+* Service Invocation - is used to invoke the sentiment analysis API in the Processor application defined later on.
+* State Store - is used to save tweets to Redis, as defined in [provider/components/statestore.yaml](provider/components/statestore.yaml)
+* Pub Sub - is used to publish tweets to be consumed by the Web UI. PubSub is defined in [provider/components/pubsub.yaml](provider/components/pubsub.yaml)
+
+The controller is implemented in [ApplicationController.java](provider/src/main/java/io/dapr/apps/twitter/processor/ApplicationController.java). It receives a tweet via input binding, then uses Service Invocation to get the sentiment analysis for the Tweet's text, saves the analyized tweet into the state store and, finally, publishes it to a PubSub topic.
+
+### Processor
+
+It will receive a text and a language code and invoke Azure's Text Analytics to extract the sentiment and score for the text. If the analysis cannot be done, it will return `unknown` sentiment with a score of `0`. It only uses secret store from Dapr's building blocks, as seen in [processor/components/secretstore.yaml](processor/components/secretstore.yaml).
```
Comment (mchmarny) [inline twitter-sentiment-processor/demos/javademo/README.md:70]: same here, "it", is it the processor service?

File: twitter-sentiment-processor/demos/javademo/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+### Viewer
+
+It served a static website from the [viewer/src/main/resources/static](viewer/src/main/resources/static) folder. The website stablished a Web Socket connection to the `/ws/` endpoint. The server side for the Web Socket is handled by [SocketTextHandler.java](viewer/src/main/java/io/dapr/apps/twitter/viewer/SocketTextHandler.java). The PubSub subscription is handled by [ApplicationController.java](viewer/src/main/java/io/dapr/apps/twitter/processor/ApplicationController.java), using a singleton instance of [WebSocketPubSub.java](viewer/src/main/java/io/dapr/apps/twitter/processor/WebSocketPubSub.java) to send the topic's message to all active Web Sockets.
```
Comment (mchmarny) [inline twitter-sentiment-processor/demos/javademo/README.md:76]: "served"

File: twitter-sentiment-processor/demos/javademo/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+## Running
+
+First, run the processor app, using `sentiment_processor` as the app id:
+```sh
+cd processor
+mvn clean install
+dapr run --app-id sentiment_processor --components-path=./components/ --app-port=8081 -- java -jar target/app.jar
+```
+
+On another terminal, run the provider app:
```
Comment (mchmarny) [inline twitter-sentiment-processor/demos/javademo/README.md:87]: "In" another terminal

File: twitter-sentiment-processor/demos/javademo/processor/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+WORKDIR /build
+
+COPY pom.xml .
+RUN mvn dependency:go-offline
+
+ADD src/ /build/src/
+RUN mvn package
+
+# package stage
+FROM openjdk:11-jre-slim
```
Comment (mchmarny) [inline twitter-sentiment-processor/demos/javademo/processor/Dockerfile:14]: Consider using distroless images https://github.com/GoogleContainerTools/distroless/blob/master/examples/java/Dockerfile

File: twitter-sentiment-processor/demos/javademo/provider/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+WORKDIR /build
+
+COPY pom.xml .
+RUN mvn dependency:go-offline
+
+ADD src/ /build/src/
+RUN mvn package
+
+# package stage
+FROM openjdk:11-jre-slim
```
Comment (mchmarny) [inline twitter-sentiment-processor/demos/javademo/provider/Dockerfile:14]: consider the use of distrolless images  https://github.com/GoogleContainerTools/distroless/blob/master/examples/java/Dockerfile

**Review by mchmarny — COMMENTED** [review]
Made a few readme and dockerfile comments. Can't speak too much to the Java code

File: twitter-sentiment-processor/demos/javademo/processor/Dockerfile (hunk 5 above)
Comment (artursouza) [reply twitter-sentiment-processor/demos/javademo/processor/Dockerfile]: Done.

File: twitter-sentiment-processor/demos/javademo/provider/Dockerfile (hunk 6 above)
Comment (artursouza) [reply twitter-sentiment-processor/demos/javademo/provider/Dockerfile]: Done.

**Comment by artursouza** [comment]
> Made a few readme and dockerfile comments. Can't speak too much to the Java code

You could just say that Java is the most beautiful language ever and my code is pure art :)

**Review by mchmarny — APPROVED** [review]
/lgtm

**Comment by artursouza** [comment]
@orizohar @msfussell @mchmarny I've merged my code. I can move it somewhere else if we decide to reorganize the twitter demo.

**Pull request merged**